client/my-sites/site-settings/publishing-tools/hooks/use-switch-post-by-voice-mutation.tsx

```diff
@@ -0,0 +1,52 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import {
+	getCachePostByVoiceKey,
+	getPostByVoicePath,
+	parsePostByVoiceResponse,
+} from 'calypso/my-sites/site-settings/publishing-tools/hooks/use-get-post-by-voice';
+import type {
+	PostByVoice,
+	PostByVoiceResponse,
+} from 'calypso/my-sites/site-settings/publishing-tools/types';
+
+export const useSwitchPostByVoiceMutation = ( siteId: number | null ) => {
```

Comment:
```suggestion
export const useTogglePostByVoiceMutation = ( siteId: number | null ) => {
```
I think `toggle` is a more intuitive term than `switch` here